src/natcap/invest/hra.py

```diff
@@ -1,3253 +1,2424 @@
+            will be either 'Euclidean' or 'Multiplicative'.
+        args['decay_eq'] (str): a string identifying the equation that should
+            be used in calculating the decay of stressor buffer influence. This
+            can be 'None', 'Linear', or 'Exponential'.
+        args['aoi_vector_path'] (str): a path to the shapefile containing one
+            or more planning regions used to get the average risk value for
+            each habitat-stressor combination over each area. Optionally, if
+            each of the shapefile features contain a 'name' field, it will
+            be used as a way of identifying each individual shape.
+        args['n_overlapping_stressors'] (number): If provided, this
+            number will be used in risk reclassification instead of the
+            calculated maximum number of stressor layers that overlap.
+        args['n_workers'] (int): the number of worker processes to
+            use for processing this model.  If omitted, computation will take
+            place in the current process. (optional)
+        args['visualize_outputs'] (bool): if True, create output GeoJSONs and
+            save them in a visualization_outputs folder, so users can visualize
+            results on the web app. Default to True if not specified.
+            (optional)
+
+    Returns:
+        None.
+
+    """
+    intermediate_dir = os.path.join(args['workspace_dir'],
+                                    'intermediate_outputs')
+    output_dir = os.path.join(args['workspace_dir'], 'outputs')
+    taskgraph_working_dir = os.path.join(args['workspace_dir'], '.taskgraph')
+    utils.make_directories([intermediate_dir, output_dir])
+    suffix = utils.make_suffix_string(args, 'results_suffix')
+
+    resolution = float(args['resolution'])
+    max_rating = float(args['max_rating'])
+    max_n_stressors = float(args['n_overlapping_stressors'])
+    target_srs_wkt = pygeoprocessing.get_vector_info(
+        args['aoi_vector_path'])['projection_wkt']
+
+    if args['risk_eq'].lower() == 'multiplicative':
+        max_pairwise_risk = max_rating * max_rating
+    elif args['risk_eq'].lower() == 'euclidean':
+        max_pairwise_risk = math.sqrt(
+            ((max_rating - 1) ** 2) + ((max_rating - 1) ** 2))
+    else:
+        raise ValueError(
+            "args['risk_eq'] must be either 'Multiplicative' or 'Euclidean' "
+            f"not {args['risk_eq']}")
+
+    try:
+        n_workers = int(args['n_workers'])
+    except (KeyError, ValueError, TypeError):
+        # KeyError when n_workers is not present in args
+        # ValueError when n_workers is an empty string.
+        # TypeError when n_workers is None.
+        n_workers = -1  # single process mode.
+    graph = taskgraph.TaskGraph(taskgraph_working_dir, n_workers)
+
+    # parse the info table and get habitats, stressors
+    habitats, stressors = _parse_info_table(args['info_table_path'])
```

Comment:
Yeah, that's a good point.  How about a rename to `habitats_info` and `stressors_info`, to clarify that these are not just sequences of strings?

In case that's a good solution, implemented in 85229f68a